src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/TaskMethodInvoker.cs

```diff
@@ -270,7 +274,9 @@ private void EnsureIsInitialized()
             {
                 // Only pass locals byref because InvokerUtil may store temporary results in the byref.
                 // If two threads both reference this.count, temporary results may interact.
-                InvokeDelegate invokeDelegate = InvokerUtil.GenerateInvokeDelegate(TaskMethod, out int inputParameterCount, out int outputParameterCount);
+                using var scope = _serviceProvider.CreateScope();
+                ILogger<InvokeDelegate> logger = scope.ServiceProvider.GetRequiredService<ILogger<InvokeDelegate>>();
```

Comment:
I'm curious why you are creating a DI scope here? I wouldn't expect a logger to require a scope. Is it to make sure Dispose() is called on the logger if needed?